packages/@aws-cdk/aws-ec2/test/test.vpc.ts

```diff
@@ -228,6 +260,20 @@ export = {
                 DestinationCidrBlock: '0.0.0.0/0',
                 NatGatewayId: { },
             }));
+            expect(stack).notTo(haveResource("AWS::EC2::Subnet",
+                hasTags([
+                    { Key: 'AddedTag', Value: 'NewThing' },
+                ])
+            ));
+            (myVpc.publicSubnets as VpcSubnet[]).forEach((subnet) => {
```

Comment:
I wanted to change `vpc.publicSubnets` to be type `VpcSubnet` instead of `VpcSubnetRef` because that made a lot of sense based on both file location and the change I am making. However, this breaks part of the java test cases. Is that an indication of an agreed to convention that is for now required or is that open for debate?

Reply:
`publicSubnets` is part of the API of `VpcNetworkRef` which can represent either a VPC defined in your stack or an [`imported`](https://awslabs.github.io/aws-cdk/refs/_aws-cdk_aws-ec2.html?highlight=vpcnetworkref#@aws-cdk/aws-ec2.VpcNetworkRef.import) VPC, which is a very important use case (see #506). Therefore, we also want the subnets to use the `XxxRef` version, so they represent a more abstract notion.

There's a [little bit of documentation](https://awslabs.github.io/aws-cdk/aws-construct-lib.html#imports) about imports under the AWS Construct Library topic in our docs. We should probably improve it...

Reply:
RE: Imports

I was under the impression (from reading the Importing section of https://awslabs.github.io/aws-cdk/refs/_aws-cdk_aws-certificatemanager.html) that you had to export a construct before you could import it

Reply:
`XxxRef.import()` static methods accept a set of props. If you can satisfy them, you can instantiate a `Ref`. For example, [`BucketRef.import`](https://awslabs.github.io/aws-cdk/refs/_aws-cdk_aws-s3.html?highlight=bucketref#@aws-cdk/aws-s3.BucketRefProps) asks for two (optional) attributes: `bucketName` and `bucketArn`. If you provide one of them, the other can be deduced, so:

```ts
const importedBucket = Bucket.import(this, 'MyImportedBucket', { bucketName: 'my_bucket' });
```

Would return a `BucketRef` instance associated with an existing bucket named `my_bucket`.

Other resources may have different heuristics for imports (for example, VPCs need much more information in order to be usable).

When you `export` something, it will produce a set of `Output`s and return a set of `Fn::ImportValue` tokens that match the same set of props needed for import, so you can export/import across stack boundaries:

```ts
const exportedFoo = fooBucket.export();

// then in another stack:
const importedFoo = Bucket.import(this, 'ImportedFoo', exportedFoo);
```

Should we expand on the [this topic](https://awslabs.github.io/aws-cdk/aws-construct-lib.html#imports)?

Reply:
I'll take a stab at updating the topic. Watch for a new PR soon.